pandas/_libs/lib.pyx

```diff
@@ -2362,9 +2366,9 @@ def maybe_convert_objects(ndarray[object] objects,
     convert_interval : bool, default False
         If an array-like object contains only Interval objects (with matching
         dtypes and closedness) or NaN, whether to convert to IntervalArray.
-    convert_to_nullable_integer : bool, default False
-        If an array-like object contains only integer values (and NaN) is
-        encountered, whether to convert and return an IntegerArray.
+    convert_to_nullable_dtype : bool, default False
+        If an array-like object contains only integer or boolean values (and NaN) is
+        encountered, whether to convert and return an Boolean/IntegerArray.
```

Comment:
Just to confirm `FloatArray` isn't supported yet?

Reply:
This is something I raised in the sql pr as well. Currently, we only use ea, when the regular dtype can’t hold missing values. So there is no real use case for FloatArray.

it would probably make more sense to always convert when this flag is set, but this is out of scope here

Reply:
Yeah. I would opt to returning `Float64` when people specify `use_nullable_dtypes=True`, especially once (hopefully) other core devs find it useful to store `NaN` & `NA`

But sure can be a follow up